iceoryx_utils/include/iceoryx_utils/internal/cxx/method_callback.inl

```diff
@@ -0,0 +1,235 @@
+// Copyright (c) 2020 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef IOX_UTILS_CXX_METHOD_CALLBACK_INL
+#define IOX_UTILS_CXX_METHOD_CALLBACK_INL
+
+#include "iceoryx_utils/cxx/method_callback.hpp"
+namespace iox
+{
+namespace cxx
+{
+namespace internal
+{
+template <typename ReturnValue>
+struct ReturnSuccess
+{
+    template <typename Callback, typename... Args>
+    static success<ReturnValue> call(Callback& callback, Args&&... args) noexcept
+    {
+        return success<ReturnValue>(callback(std::forward<Args>(args)...));
+    }
+};
+
+template <>
+struct ReturnSuccess<void>
+{
+    template <typename Callback, typename... Args>
+    static success<void> call(Callback& callback, Args&&... args) noexcept
+    {
+        callback(std::forward<Args>(args)...);
+        return success<void>();
+    }
+};
+
+template <typename ReturnValue, typename ClassType, typename... Args>
+ReturnValue
+constMethodCallbackCaller(void* classPtr, ReturnValue (GenericClass::*methodPtr)(Args...) const, Args&&... args)
```

Comment:
better: objectPtr or similar, this is often more appropriate in many other places (I will not mark those)

Reply:
fixed